axum/src/json.rs

```diff
@@ -112,6 +113,27 @@ where
     }
 }
 
+impl<T, S> OptionalFromRequest<S> for Json<T>
+where
+    T: DeserializeOwned,
+    S: Send + Sync,
+{
+    type Rejection = JsonRejection;
+
+    async fn from_request(req: Request, state: &S) -> Result<Option<Self>, Self::Rejection> {
+        if json_content_type(req.headers()) {
+            let bytes = Bytes::from_request(req, state).await?;
+            if !bytes.is_empty() {
+                Ok(Some(Self::from_bytes(&bytes)?))
+            } else {
+                Ok(None)
```

Comment:
if the header says that the content-type is JSON, but the body is empty, shouldn't that be an error?

Reply:
yeah that makes sense, I added a new error, is that the right way to approach this?

Reply:
what rejection is the non-optional extractor using in this case? I assume `JsonSyntaxError`? would it be possible to reuse that rejection?

Reply:
I looked into that one, I am not quite sure what `Error` is supposed to be here in `JsonSyntaxError(Error)` ? Is it really just specific to serde?

Reply:
I guess we should always call `Self::from_bytes()` even if `bytes.is_empty()`, because if the header was set then we expect valid JSON and an empty body is not valid JSON 🤔

Reply:
do i need to create a deserializer and then use `serde_path_to_error::deserialize` to use that error variant?

Reply:
> I guess we should always call Self::from_bytes() even if bytes.is_empty(), because if the header was set then we expect valid JSON and an empty body is not valid JSON 🤔

something like:

```rs
match Self::from_bytes(&bytes) {
  Ok(_) => unreachable!("JSON body can only be empty here"),
  Err(err) => Err(err),
}               
```